hibernate-core/src/main/java/org/hibernate/metamodel/internal/AbstractPojoInstantiator.java

```diff
@@ -34,4 +34,10 @@ public boolean isAbstract() {
 	public boolean isInstance(Object object, SessionFactoryImplementor sessionFactory) {
 		return mappedPojoClass.isInstance( object );
 	}
+
+	@Override
+	public boolean isSameClass(Object object, SessionFactoryImplementor sessionFactory) {
+		return object.getClass() == mappedPojoClass;
```

Comment:
Should we take consideration the possibility that `object` could be null? This is a public method so it should be function appropriately for any external input. Nullness processing is always an gray area in Hibernate's codebase, IMHO.

Reply:
it is an internal class, plus returning `false` seems wrong to me so in any case if `object` is null an exception should be raised.